ingestion/src/kldata.rs

```diff
@@ -35,47 +44,84 @@ struct ObsinnId {
 
-    let res = value
-        .parse::<T>()
-        .map_err(|_| Error::Parse(format!("invalid number in kldata header for key {}", key)))?;
+        let kldata_string = fields
+            .next()
+            .ok_or_else(|| Error::Parse("kldata header terminated early".to_string()))?;
 
-    Ok(res)
-}
+        if kldata_string != "kldata" {
+            return Err(Error::Parse(
+                "kldata indicator missing or out of order".to_string(),
+            ));
+        }
 
-fn parse_meta(meta: &str) -> Result<(i32, i32, usize), Error> {
-    let mut parts = meta.splitn(4, '/');
+        let unexpected_field = |field: &str| {
+            Error::Parse(format!(
+                "unexpected field in kldata header format: {}",
+                field
+            ))
+        };
 
-    let next_err = || Error::Parse("kldata header terminated early".to_string());
+        let mut header = ObsinnHeader::default();
 
-    if parts.next().ok_or_else(next_err)? != "kldata" {
-        return Err(Error::Parse(
-            "kldata indicator missing or out of order".to_string(),
-        ));
-    }
+        for field in fields.by_ref() {
+            // TODO: this field signals data deletion/update in kvalobs, we do not use it
+            if field == "add" {
+                continue;
+            }
+
+            let (key, value) = field
+                .split_once('=')
+                .ok_or_else(|| unexpected_field(field))?;
+
+            // TODO: check possible ordering by logging incoming messages
+            match key {
+                "nationalnr" => header.station_id = Some(parse_value(key, value)?),
+                "type" => header.type_id = Some(parse_value(key, value)?),
+                "received_time" => header.received_time = Some(parse_value(key, value)?),
+
+                // TODO: maybe we should also check if messageid is present
+                "messageid" => header.message_id = parse_value(key, value)?,
+                _ => return Err(unexpected_field(field)),
+            }
+        }
 
-    let station_id = parse_meta_field(parts.next().ok_or_else(next_err)?, "nationalnr")?;
-    let type_id = parse_meta_field(parts.next().ok_or_else(next_err)?, "type")?;
-    let message_id = parse_meta_field(parts.next().ok_or_else(next_err)?, "messageid")?;
+        if header.station_id.is_none() {
+            return Err(Error::Parse(
+                "missing field `nationalnr` in kldata header".to_string(),
+            ));
+        }
 
-    Ok((station_id, type_id, message_id))
+        if header.type_id.is_none() {
+            return Err(Error::Parse(
+                "missing field `type` in kldata header".to_string(),
+            ));
+        }
+
+        Ok(header)
+    }
+}
+
+fn parse_value<T: FromStr>(key: &str, value: &str) -> Result<T, Error>
+where
+    <T as FromStr>::Err: Debug,
+{
+    value
+        .parse::<T>()
+        .map_err(|_| Error::Parse(format!("invalid number in kldata header for key {}", key)))
```

Comment:
Nitpick, but I think "number" here is now technically wrong, since you're using this to parse strings now for received_time